specs/phase0/validator.md

```diff
@@ -606,15 +609,28 @@ def get_aggregate_and_proof_signature(state: BeaconState,
 
 ## Phase 0 attestation subnet stability
 
-Because Phase 0 does not have shards and thus does not have Shard Committees, there is no stable backbone to the attestation subnets (`beacon_attestation_{subnet_id}`). To provide this stability, each validator must:
+Because Phase 0 does not have shards and thus does not have Shard Committees, there is no stable backbone to the attestation subnets (`beacon_attestation_{subnet_id}`). To provide this stability, each beacon node should:
 
-* Randomly select and remain subscribed to `RANDOM_SUBNETS_PER_VALIDATOR` attestation subnets
-* Maintain advertisement of the randomly selected subnets in their node's ENR `attnets` entry by setting the randomly selected `subnet_id` bits to `True` (e.g. `ENR["attnets"][subnet_id] = True`) for all persistent attestation subnets
-* Set the lifetime of each random subscription to a random number of epochs between `EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION` and `2 * EPOCHS_PER_RANDOM_SUBNET_SUBSCRIPTION]`. At the end of life for a subscription, select a new random subnet, update subnet subscriptions, and publish an updated ENR
+* Remain subscribed to `SUBNETS_PER_NODE` for `SUBNET_DURATION_IN_EPOCHS` epochs.
+* Maintain advertisement of the selected subnets in their node's ENR `attnets` entry by setting the selected `subnet_id` bits to `True` (e.g. `ENR["attnets"][subnet_id] = True`) for all persistent attestation subnets.
+* Select these subnets based on their node-id as specified by the following `compute_subnets(node_id,epoch)` function.
 
-*Note*: Short lived beacon committee assignments should not be added in into the ENR `attnets` entry.
+```python
+def compute_subnet(node_id: int, epoch: Epoch, index: int) -> int:
+    node_id_prefix = node_id >> (256 - ATTESTATION_SUBNET_PREFIX_BITS)
+    permutation_seed = hash(uint_to_bytes(epoch // SUBNET_DURATION_IN_EPOCHS))
```

Comment:
If you made this

```python
node_offset = node_id % SUBNET_DURATION_IN_EPOCHS
permutation_seed = hash(uint_to_bytes((epoch + node_offset))// SUBNET_DURATION_IN_EPOCHS))
```

then this would stagger the subnet churn but still be deterministic.  It would also mean that there wouldn't be complete "dump and subscribe" periods, as there would always be some nodes still subscribed to each subnet as other nodes are transitioning from one subnet to another.

Reply:
I think the issue with this is that there would be periods of time where no nodes are subscribed to an individual subnet (or at least not searchable or enforceable). 
I did look at the staggering, and I found it difficult to maintain a uniform distribution of nodes on subnets for all epochs.